src/bloqade/task/braket_simulator.py

```diff
@@ -38,14 +74,42 @@ def run(self, **kwargs) -> "BraketEmulatorTask":
         self.task_result_ir = from_braket_task_results(aws_task.result())
         return self
 
-    def result(self):
+    def result(self) -> QuEraTaskResults:
+        """
+        Retrieves the results of the task.
+
+        Returns:
+            QuEraTaskResults: The results of the task.
+
+        Raises:
+            ValueError: If the task has not been submitted yet.
+
+        Example:
+            >>> task_spec = BraketTaskSpecification(...)  # Task specification setup
```

Comment:
this needs to be more specific, e.g what does the specification take? and I don't think this is what user will be using. The result is called in the builder call chain.